### ISSUE TITLE
Mouse coords are wrong on mac

### ISSUE THREAD

**Issue by joshmarinacci** [creationix/node-sdl#9]
Note: This may affect other platforms as well, but I'm currently working on mac so that's all I've tested.

If you move across the screen with just mouse presses you will see that the coordinates don't update with each press. they are still the same as the first press. However, if you press, drag, and release, then the coordinates are updated. I suspect there is some broken logic in the C++ side of things.

**Comment by joshmarinacci** [comment]
I've discovered the root issue but I'm not sure how to fix it.

A plain C app using the most recent SDL receives the proper mouse coordinates, but a Node version does not.  The root cause appears to be how the Cocoa layer is initialized.  In the C version you first run an SDLMain.m which invokes your app's main. Along the way it does a bunch of Cocoa setup and starts the Cocoa event loop with [NSApp run].  Node does a quick [NSApplication getSharedInstance] but does not do the rest of the setup.  Without the full Cocoa setup lots of things are broken, like the menubar, system menu, dock icon, etc.  as well as many facets of the mouse events.  

I couldn't figure out how to modify sdl.cc to properly setup cocoa and still get it all to compile with node-waf or node-gyp. However, using the NodObjC module I was able to get a plain cocoa app up and running.  Combining the two I can now get proper mouse events.

There is a downside though. Going the full Cocoa route seems to put Cocoa in charge of the main event loop.  setInterval no longer works. It simply never calls the callback function.  That's going to make it hard to do any kind of animation.
